manifest/oci_index.go

```diff
@@ -89,11 +98,20 @@ func (index *OCI1Index) ChooseInstance(ctx *types.SystemContext) (digest.Digest,
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				matchedDigest = d.Digest
```

Comment:
AFAICS this changes behavior: before were using the first match from equal (non-std ones), now we are using the last one.

The original behavior needs to be preserved (it’s essential to the design of the compatibility mechanism).

Reply:
Matched now and added a test.